FirebaseSessions/Sources/Settings/SDKDefaultSettings.swift

```diff
@@ -0,0 +1,46 @@
+//
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+
+/// Class that manages the local overrides configs related to the library.
+class SDKDefaultSettings: SettingsProvider, SettingsProtocol {
+  var sessionsEnabled: Bool? {
+    // Default is sessions enabled
+    return true
+  }
+
+  var sessionTimeout: TimeInterval? {
+    // Default is 30 minutes
+    return 30 * 60
+  }
+
+  var samplingRate: Double? {
+    // Default is all events are dispatched
+    return 1.0
+  }
+}
+
+typealias SDKDefaultSettingsProvider = SDKDefaultSettings
```

Comment:
Not action needed, just curious what's this `typealias` + `extension` pattern, is it a recommended pattern in swift?

Reply:
Typealias is a technique to make the code more modular and readable. In this example, I have split the big class into sub modules depending on the action they perform (implementing different protocols). As classes grow, this is a simple yet powerful technique to make the code maintainable.

Some references: https://www.programiz.com/swift-programming/typealias, https://www.avanderlee.com/swift/typealias-usage-swift/

Happy to provide a lot more context offline.